### DISCUSSION TITLE
more intuitive new SQL tab

### DISCUSSION THREAD

**Discussion by rdp** [dbeaver/dbeaver#13420] [Ideas]
When I first tried out dbeaver, I could traverse database table tree hierarchy, and modify data there, but it wasn't clear how to pull open a new "SQL query" tab.  If you right click near the tabs the only option is "close" or "detach" no "open new tab" or anything.
<img width="521" alt="Screen Shot 2019-07-25 at 9 41 01 AM" src="https://user-images.githubusercontent.com/12496/61888484-cbf07c80-aec0-11e9-8cd5-832e54410354.png">

Appears I'm not the only one who gets confused by this: https://stackoverflow.com/q/34526817/32453

Wonder if there's a way to make it more intuitive here.  Thanks!

**Comment by cerebel** [comment]
You have in main menu 'special' item "SQL Editor"  (and two additional icons in Toolbar for sql editor).
https://raw.githubusercontent.com/wiki/dbeaver/dbeaver/images/ug/appwindow-with-markup.png

**Comment by rdp** [comment]
Hmm OK those buttons on the toolbar work.  But it's not immediately evident to a beginner what they do, and it takes actually awhile for the tooltip to appear to explain it.  A couple more options that might help for the newbie: having the "tooltip" appear for the toolbar more quickly.  Or change the icon so it includes "SQL" somehow.  And/or having a right click menu on the "connection" have an option "new SQL" that type of thing... (update: appears the connection itself does have the option, but not the database if you right click on it, which somehow confused me, FWIW...)

**Comment by rdp** [comment]
I think the general gist of this can be boiled down to "the SQL Editor toolbar mini icon" is...confusing...it looks like little snakes or something eating one another?  At least for me I didn't know what it meant initially...

**Comment by p91paul** [comment]
This was moved to discussion, but with the new sql script design I believe it can be safely closed. It's actually doing what the OP wanted (SQL text next to the icon + descriptions in the dropdown menu)

**Comment by rdp** [comment]
Thanks!

On Tuesday, August 3, 2021, Paolo Inaudi ***@***.***> wrote:
> This was moved to discussion, but with the new sql script design I
believe it can be safely closed. It's actually doing what the OP wanted
(SQL text next to the icon + descriptions in the dropdown menu)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
> Triage notifications on the go with GitHub Mobile for iOS or Android.<
https://ci3.googleusercontent.com/proxy/r5yBWn96RBQVPOJpx4rIp2vnsBNL8r9yeU_4TMBjgMuVc7i1x09dSjon12C3K50UfQiLGcczzxnQuh2f1edKRDg9OtjyCm64ouAAofPl_eWY1qzd0sZq5ThGhxPGsCnJ-2gIjVoFpoiCxKU2-SyiTmow8dQjtGnnAVD55mgquhFXZAh9-DnQ9BeIJ20mhhtjIXH-fHbYAtLKT272t_lWJu9B1V51KVQ_KJvRJoBP4dDd1A3DBiTO=s0-d-e1-ft#https://github.com/notifications/beacon/AAADBUDUS2OHAWKIRGLSPM3T3A7GDA5CNFSM5BPOYYPKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZGOAAITF3A.gif>

**Comment by rdp** [comment]
Yep looks grand!
